tests/specs/nodeRewardsProgramModule.spec.js

```diff
@@ -0,0 +1,535 @@
+
+            spyOn(Alert, 'addressEnrolled');
+            spyOn(Alert, 'invalidEnrollmentAddress');
+
+            const ctrl = $controller('NodeRewardsProgramCtrl', {
+                $scope: $rootScope.$new()
+            });
+
+            // Act:
+            await ctrl.sendEnroll();
+
+            $timeout.flush();
+
+            // Assert:
+            expect(Alert.addressEnrolled).toHaveBeenCalled();
+            expect(Alert.invalidEnrollmentAddress).toHaveBeenCalled();
+            expect(ctrl.okPressed).toEqual(false);
+            done();
+        });
+
+        it('returns transaction object and announce transaction', async (done) => {
+            // Arrange:
+            spyOn(Wallet, 'decrypt').and.returnValue(true);
+            spyOn(Wallet, 'transact');
+
+            spyOn(NodeRewardsProgram, 'checkEnrollmentStatus').and.returnValue(Promise.resolve(false));
+            spyOn(NodeRewardsProgram, 'checkEnrollmentAddress').and.returnValue(Promise.resolve(true));
+            spyOn(NodeRewardsProgram, 'getCodewordHash').and.returnValue(Promise.resolve('hash'));
+
+            const ctrl = $controller('NodeRewardsProgramCtrl', {
+                $scope: $rootScope.$new()
+            });
+
+            spyOn(ctrl, 'prepareTransaction');
+
+            // Act:
+            await ctrl.sendEnroll();
+
+            $timeout.flush();
+
+            // Assert:
+            expect(Wallet.transact).toHaveBeenCalled();
+            expect(ctrl.prepareTransaction).toHaveBeenCalled();
+            expect(ctrl.okPressed).toEqual(false);
+            done();
+        });
+    })
+
+    describe('setEnroll', () => {
+        it('set IP or domain to empty when accounts have not enroll', async (done) => {
+            // Arrange:
+            spyOn(NodeRewardsProgram, 'getNodeInfo').and.returnValue(Promise.reject('error'));
+
+            const ctrl = $controller('NodeRewardsProgramCtrl', {
+                $scope: $rootScope.$new()
+            });
+
+            await ctrl.getStatus();
+            $timeout.flush();
+
+            // Act:
+            ctrl.setEnroll();
+
+            // Assert:
+            expect(ctrl.formData.domainOrIP).toEqual('');
+            expect(ctrl.preparedTransaction.type).toEqual(nem.model.transactionTypes.transfer);
+            expect(ctrl.preparedTransaction.fee).toEqual(200000);
+            expect(ctrl.preparedTransaction.message.payload).toEqual(nem.utils.convert.utf8ToHex(`enroll ${ctrl.formData.domainOrIP} ${'0'.repeat(64)}`));
+            done();
+        })
+
+        it('set IP or domain to exist endpoint when accounts enrolled', async (done) => {
+            // Arrange:
+            spyOn(NodeRewardsProgram, 'getNodeInfo').and.returnValue(Promise.resolve(mockEnrollStatus));
+
+            const ctrl = $controller('NodeRewardsProgramCtrl', {
+                $scope: $rootScope.$new()
+            });
+
+            await ctrl.getStatus();
+            $timeout.flush()
```

Comment:
missing semicolons, if possible would be good to run linter on new/changed files

Reply:
the current project didn't have a lint setup.
should I apply our current lint setup?

Reply:
> should I apply our current lint setup

i think it would be good to do for the new files with one modification: current lint setup expects tab indents whereas NanoWallet is using spaces. it's a mess to mix tab and space indented files in single repo, so i'd adjust rules to expect space indentation before running.